File: datasets/IR-Plag/convertai-original-java-rest-kotlin/case-07/plagiarized/L1/08/Level1.kt
package IR

import java.util.Scanner

/**
 *
 * @author FD8DBE9073AFCC0504CD6901E1091CAD
 */
fun main() {
    val input = Scanner(System.`in`) //harus import dulu
    println("Enter a 4 by 4 matrix row by row: ")
    val m = Array(4) { DoubleArray(4) }
    for (i in 0 until 4) {
        for (j in 0 until 4) {
            m[i][j] = input.nextDouble()
        }
    }
    println("Sum of the elements in the major diagonal is ${sumMajorDiagonal(m)}")
}

//fungsi untuk menjumlahkan diagonal
fun sumMajorDiagonal(m: Array<DoubleArray>): Double {
    var sum = 0.0
    for (i in 0 until m.size) {
        sum += m[i][i]
    }
    return sum
}

File: datasets/IR-Plag/convertai-original-java-rest-python/case-07/plagiarized/L1/08/Level1.py
import numpy as np

def sum_major_diagonal(m):
    """
    Function to calculate the sum of the elements in the major diagonal of a matrix.
    
    Args:
        m (numpy.ndarray): A 4x4 matrix.
    
    Returns:
        float: The sum of the elements in the major diagonal.
    """
    return np.trace(m)

if __name__ == "__main__":
    # Get input matrix from the user
    print("Enter a 4 by 4 matrix row by row:")
    m = np.array([list(map(float, input().split())) for _ in range(4)])
    
    # Calculate and print the sum of the major diagonal
    print(f"Sum of the elements in the major diagonal is {sum_major_diagonal(m)}")

